docs/K8s-test-plan.md

```diff
@@ -0,0 +1,362 @@
+
+### TC_LOCAL_KUBE_1: Switch between Local Mode and Kube Mode
+#### Test Objective
+Verify Device Under Test (DUT) properly transitions between local mode and kube mode when manifest is properly applied.
+#### Test Configuration
+- Pick up from step 6 of [TC_JOIN_1](#tc_join_1-join-master-once-available)
+- {feature x} `current_owner=local`, `set_owner=kube`, `remote_state=none`, `container_version=1.0.0`
+#### Test Steps
+1. Apply manifest for {feature x} v1.1.1 with valid URL
+   - local {feature x} v1.0.0 container should terminate
+   - k8s {feature x} v1.1.1 container from remote image should run
+   - **Expect:** `current_owner=kube`, `container_version=1.1.1`, `remote_state=running`
+2. Set {feature x} desired owner to local (`set_owner=local`)
+   - k8s {feature x} v1.1.1 container from remote image should terminate
+   - local {feature x} v1.0.0 container should run
+   - **Expect:** `current_owner=local`, `container_version=1.0.0`, `remote_state=none`
+3. Set {feature x} desired owner to kube (`set_owner=kube`)
+   - local {feature x} v1.0.0 container should terminate
+   - k8s {feature x} v1.1.1 container from remote image should run
+   - **Expect:** `current_owner=kube`, `container_version=1.1.1`, `remote_state=running`
+
+### TC_LOCAL_KUBE_2: Upgrade Kube Feature v1.1.1 to Kube Feature v2.0.0 via Successful Manifest Application
+#### Test Objective
+Verify Device Under Test (DUT) running kube mode feature container v1.1.1 properly upgrades to kube mode feature v2.0.0 upon successful application of v2.0.0 manifest.
+#### Test Configuration
+- Pick up from step 3 of [TC_LOCAL_KUBE_1](#tc_local_kube_1-switch-between-local-mode-and-kube-mode)
+- {feature x} `current_owner=kube`, `set_owner=kube`, `remote_state=running`, `container_version=1.1.1`
+#### Test Steps
+1. Deploy {feature x} v2.0.0 manifest with valid URL
+   - New k8s {feature x} v2.0.0 container should start running
+   - {feature x} v1.1.1 container will continue running until master disables v1.1.1
+   - **Expect:** `container_version=2.0.0`, `remote_state=running`
+2. Stop {feature x} service
+   - k8s {feature x} v2.0.0 container should stop running
+   - **Expect:** `container_version=2.0.0`, `remote_state=stopped`
+3. Start {feature x} service
+   - k8s {feature x} v2.0.0 container should start running
+   - **Expect:** `container_version=2.0.0`, `remote_state=running`
+
+### TC_LOCAL_KUBE_3: Local to Kube Feature with Failed Manifest Application
+#### Test Objective
+Verify Device Under Test (DUT) local mode feature properly responds to `set_owner=kube` when kube mode feature fails to deploy properly (failed manifest application due to invalid URL). Local mode container should keep running until kube mode feature is successfully deployed.
+#### Test Configuration
+- Pick up from step 6 of [TC_JOIN_1](#tc_join_1-join-master-once-available)
+- {feature x} `current_owner=local`, `set_owner=kube`, `remote_state=none`, `container_version=1.0.0`
```

Comment:
Can you add a section before the steps to describe how to change the configuration, how to check the DUT and feature status?

Reply:
Yes, I have added this to the description of the statuses and configurations. Please let me know if I should adjust the placement